File: Backend/Services/ExamService/OnlineExamSystem.Exam.Domain/Entities/Section.cs
using OnlineExamSystem.Exam.Domain.Enums;
using OnlineExamSystem.Shared.Common.Entities;

namespace OnlineExamSystem.Exam.Domain.Entities;

public class Section : BaseEntity
{
    public Guid ExamId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Instructions { get; set; } = string.Empty;
    public int DisplayOrder { get; set; }

    // Admin-entered targets, not auto-synced to however many questions actually get
    // assigned - same precedent as ExamPaper.TotalQuestions since Day 19.
    public int QuestionCount { get; set; }
    public int Marks { get; set; }
    public int DurationMinutes { get; set; }

    public NavigationType NavigationType { get; set; } = NavigationType.Free;
    public bool NegativeMarkingEnabled { get; set; }
    public decimal NegativeMarks { get; set; }
    public bool ShuffleQuestions { get; set; } = true;
    public bool ShuffleOptions { get; set; } = true;
    public bool AllowReview { get; set; } = true;
}

File: Frontend/examvault-web/src/types/section.ts
export type NavigationType = 'Free' | 'Sequential' | 'Locked';

export interface SectionRequest {
  name: string;
  description: string;
  instructions: string;
  displayOrder: number;
  questionCount: number;
  marks: number;
  durationMinutes: number;
  navigationType: NavigationType;
  negativeMarkingEnabled: boolean;
  negativeMarks: number;
  shuffleQuestions: boolean;
  shuffleOptions: boolean;
  allowReview: boolean;
}

export interface SectionResponse extends SectionRequest {
  id: string;
  examId: string;
  createdAtUtc: string;
}

export interface SectionOrderItem {
  sectionId: string;
  displayOrder: number;
}

// Super Admin platform-wide Sections browse only - separate shape from
// SectionResponse (adds examTitle/tenantId, drops the per-exam-only fields
// a cross-tenant browse has no use for).
export interface PlatformSectionResponse {
  id: string;
  examId: string;
  examTitle: string;
  tenantId: string;
  name: string;
  displayOrder: number;
  questionCount: number;
  marks: number;
  durationMinutes: number;
  createdAtUtc: string;
}
